model/riscv_pmp_control.sail

```diff
@@ -123,7 +152,18 @@ function pmpCheck forall 'n, 'n > 0. (addr: xlenbits, width: int('n), acc: Acces
       PMP_Continue => (),
     }
   };
-  if priv == Machine then None() else Some(accessToFault(acc))
+  if priv == Machine
+  then {
+    if sys_has_mseccfg() & mseccfg[MMWP] == 0b1                /* Make Read, Write and execute denied by default, if condition meets for M mode */
+    then Some(accessToFault(acc))
+    else if sys_has_mseccfg() & mseccfg[MML] == 0b1            /* Make execute denied, if condition meets for M mode */
+    then match acc {
+      Execute(_) => Some(accessToFault(acc)),
+      _          => None()
+    }
+    else None()
+  }
+  else Some(accessToFault(acc))
```

Comment:
I think this could be a lot simpler:

```
if priv == Machine & mseccfg[MMWP] != 0b1 & not(mseccfg[MML] == 0b1 & isExecute(acc))
then None() else Some(accesstoFault(acc)
```

(add `isExecute()` as a helper)